### ISSUE TITLE
In Multi-Select second lookup does not show list for selection

### ISSUE THREAD

**Issue by AndreasEpping** [pozil/sfdc-ui-lookup-lwc#77]
First of all, thank you for sharing this awesome lookup component. It is really a game-changer.

If I want to select multiple records from Contact or User, the first selection works pretty well. However, the second selection does not bring up the list to select a second contact or user. 
I have attached a screen recording so that you see what happens. I need to move to a different field and come back to the lookup field in order to show the list again.
I did some console log statements and from my perspective, everything is working fine. The setSearchResults is called with the newly found results. It seems that the list is simply not visible. 

https://user-images.githubusercontent.com/76778070/108830593-dce7ff00-75c9-11eb-9fcb-570e133fc8e7.mp4

**Comment by pozil** [comment]
Hi @AndreasEpping, thanks for reporting this issue with these details. At first glance I thought that it was a cache issue because of the identical search terms and object names but I can't reproduce the issue.

I'm about to release a spring '21 update this week but can you confirm that you're using the latest version of the lookup (v2.8.0)?

**Comment by AndreasEpping** [comment]
Hi @pozil ,
yes I can confirm that I am using the latest version v2.8.0. Just added some console.log statements to get behind the scenes.

**Comment by pozil** [comment]
Ok thanks for confirming that.
Are you using default search results with recent records?
I just found a probably unrelated bug while trying to reproduce your issue: the lookup may display a default search result that was already selected by the user.

**Comment by AndreasEpping** [comment]
Thanks for answering.
No I am not using default search result right now. It is really weird, because if I move the mouse to a different field and come back to the lookup, the results show up immediately. It seems that the list is simply not displayed.

**Comment by pozil** [comment]
Still no luck in reproducing the issue.
Let's try a few things:
1. can you tell me which browser you're using?
1. can you reproduce the issue with the sample app shipped with this project?
1. if not, can you share some sample code?

**Comment by AndreasEpping** [comment]
Hi Pozil,
concerning 1. I am using Google Chrome on Mac OSX newest version. But I can reproduce the same behavior with Safari.

Concerning 2. I have installed the sample app (copied the source code) but I am unable to access it in my org. I am not really used to lightning aura component development.

Concerning code.
This is the APEX controller I am using:
```
public with sharing class LwcEditTouchpointController {
    private final static Integer MAX_RESULTS = 5;
    /**
     * Searches Accounts for a specific search term
     * @param searchTerm term that is used for the search
     * @param selectedIds AccountIds that should be removed from the search results because they are already selected
     *
     * @return List of LookupSearchResults containing the found accounts
     */
    @AuraEnabled(Cacheable=true)
    public static List<LookupSearchResult> searchAttendees(String searchTerm, List<String> selectedIds) {
        // Prepare query paramters
        searchTerm += '*';

        // Execute search query
        List<List<SObject>> searchResults = [
                FIND :searchTerm
                IN ALL FIELDS
                RETURNING
                    Contact(Id, FirstName, LastName, Account.Name, Email  WHERE Id NOT IN :selectedIds),
                    User(Id, FirstName, LastName, Department, Email WHERE Id NOT IN :selectedIds),
                    Lead(Id, FirstName, LastName, Company, Email WHERE Id NOT IN :selectedIds)
                LIMIT :MAX_RESULTS
        ];

        // Prepare results
        List<LookupSearchResult> results = new List<LookupSearchResult>();

        // Extract Contacts & convert them into LookupSearchResult
        String contactIcon = 'standard:contact';
        Contact[] contacts = (List<Contact>) searchResults[0];
        for (Contact contact: contacts) {
            String title = contact.LastName + ' ' + contact.FirstName;
            String subtitle = contact.Email == null ? contact.Account.Name : contact.Email + ' • ' + contact.Account.Name;
            results.add(new LookupSearchResult(contact.Id, 'Contact', contactIcon, title, subtitle, null));
        }

        // Extract Users & convert them into LookupSearchResult
        String userIcon = 'standard:user';
        User[] users = (List<User>) searchResults[1];
        for (User user: users) {
            String title = user.LastName + ' ' + user.FirstName;
            String subtitle = user.Email + ' • ' + user.Department;
            results.add(new LookupSearchResult(user.Id, 'User', userIcon, title, subtitle, null));
        }

        // Extract Leads & convert them into LookupSearchResult
        String leadIcon = 'standard:lead';
        Lead[] leads = (List<Lead>) searchResults[2];
        for (Lead lead: leads) {
            String title = lead.LastName + ' ' + lead.FirstName;
            String subtitle = lead.Email == null ? lead.Company : lead.Email + ' • ' + lead.Company;
            results.add(new LookupSearchResult(lead.Id, 'Lead', userIcon, title, subtitle, null));
        }

        // Optionally sort all results on title
        results.sort();

        return results;
    }

    @AuraEnabled(Cacheable=true)
    public static List<LookupSearchResult> getRecentlyViewed() {
        List<LookupSearchResult> results = new List<LookupSearchResult>();
        // Get recently viewed records of type Account or Opportunity
        List<RecentlyViewed> recentRecords = [
                SELECT Id, Name, Type
                FROM RecentlyViewed
                WHERE Type = 'Contact' OR Type = 'Lead' OR Type = 'User'
                ORDER BY LastViewedDate DESC
                LIMIT 5
        ];
        // Convert recent records into LookupSearchResult
        for (RecentlyViewed recentRecord : recentRecords) {
            if (recentRecord.Type == 'Contact') {
                results.add(
                        new LookupSearchResult(
                                recentRecord.Id,
                                'Contact',
                                'standard:contact',
                                recentRecord.Name,
                                'Contact • ' + recentRecord.Name,
                                null
                        )
                );
            } else if (recentRecord.Type == 'Lead'){
                results.add(
                        new LookupSearchResult(
                                recentRecord.Id,
                                'Lead',
                                'standard:lead',
                                recentRecord.Name,
                                'Lead • ' + recentRecord.Name,
                                null
                        )
                );
            } else {
                results.add(
                        new LookupSearchResult(
                                recentRecord.Id,
                                'User',
                                'standard:user',
                                recentRecord.Name,
                                'User • ' + recentRecord.Name,
                                null
                        )
                );

            }
        }
        return results;
    }
}
```

This is the LWC controller
```
import { LightningElement, api, wire } from 'lwc';
import { ShowToastEvent } from 'lightning/platformShowToastEvent';

import searchAttendees from '@salesforce/apex/LwcEditTouchpointController.searchAttendees';
import getRecentlyViewed from '@salesforce/apex/LwcEditTouchpointController.getRecentlyViewed';

export default class EditTouchpointAttendeeLookup extends LightningElement {
    errors = [];
    recentlyViewed = [];
    _selection = [];

    /**
     * Loads recently viewed records and set them as default lookpup search results (optional)
     */
    @wire(getRecentlyViewed)
    getRecentlyViewed({ data }) {
        if (data) {
            this.recentlyViewed = data;
            this.initLookupDefaultResults();
        }
    }

    connectedCallback() {
        this.initLookupDefaultResults();
    }

    /**
     * Initializes the lookup default results with a list of recently viewed records (optional)
     */
    initLookupDefaultResults() {
        // Make sure that the lookup is present and if so, set its default results
        const lookup = this.template.querySelector('c-lookup');
        if (lookup) {
            lookup.setDefaultResults(this.recentlyViewed);
        }
    }

    // Calls APEX to find Contacts, Users or Leads
    handleSearch(event) {
        const target = event.target;
        window.console.log('--->handleSearch'+JSON.stringify(event.detail));
        searchAttendees(event.detail)
            .then(results => {
                window.console.log('Search results:' +JSON.stringify(results));
                target.setSearchResults(results);
            })
            .catch(error => {
                this.notifyUser('Lookup Error', 'An error occured while searching with the lookup field.', 'error');
                console.error('Lookup error', JSON.stringify(error));
                this.errors = [error];
            });
    }

    @api
    isValid() {
        this.checkForErrors();
        return !this.errors.length;
    }

    // User has changed the selection
    handleLookupSelectionChange(event) {
        const selection = event.target.getSelection();
        if ( selection ) {
            this._selection = selection;
        } else {
            this._selection = undefined;
        }
        this.checkForErrors();
    }

    // Displays an error message if something is wrong
    checkForErrors() {
        this.errors = [];
    }


    // Shows a toast in case an APEX callback fails
    notifyUser(title, message, variant) {
        // Notify via toast (only works in LEX)
        const toastEvent = new ShowToastEvent({ title, message, variant });
        this.dispatchEvent(toastEvent);
    }
}
```

And this is the HTML LWC code
```
<template>
    <c-lookup onsearch={handleSearch}
              errors={errors}
              label="Attendees"
              onselectionchange={handleLookupSelectionChange}
              placeholder="Search Contacts,Users and Leads"
              selection={_selection}
              is-multi-entry=true
              required={required}>
    </c-lookup>
</template>
```

**Comment by pozil** [comment]
Thanks for the sample code @AndreasEpping. I was able to reproduce the issue and fix it.
I'll release the fix in v2.9.0.
